### ISSUE TITLE
Running out of ram before training the first epoch

### ISSUE THREAD

**Issue by ngaggion** [MIC-DKFZ/nnUNet#986]
Hello, 

I'm running nnUNet on the task of 2-d images segmentation. I have already trained nnUNet for the task of chest-x-ray segmentation with ~190 training images of 1024x1024 without any problems, but I'm running into trouble when trying to run it on more bigger images. In particular, with ~130 plant root images of 3280 x 2464 (sort of equivalent to 3d images of 200x200x200).
We have 64gb of ram on our server, I believe the main issue may be that nnunet tries to locate all the images on ram for the online augmentation. Is there some way to avoid this?

Thank you for your work,
Best regards

**Comment by FabianIsensee** [comment]
nnU-Net will only try to fit entire images into RAM during planning and preprocessing. Training will only partially read the images on demand. nnU-Net will never try to read the entire dataset at once.
Can you please give some more details about the problem? What command is affected? How does it manifest? (is the RAM usage increasing steadily during training for example?)

**Comment by ngaggion** [comment]
Planning and preprocessing has no issue. The first command I run is:
````
nnUNet_plan_and_preprocess -t 601 -pl3d None --verify_dataset_integrity
````
Then the problem arises when I run:
````
nnUNet_train 2d nnUNetTrainerV2 Task601_MulticlassRootSegmentation 0 --npz
````

The ram usage starts increasing steadily till all my memory is used and the process ends abruptly. It does not finish even the first training epoch.

**Comment by FabianIsensee** [comment]
In that case the first thing you can try is to reduce the number of augmentation workers by running
`nnUNet_n_proc_DA=6 nnUNet_train 2d nnUNetTrainerV2 Task601_MulticlassRootSegmentation 0 --npz`
(the default is 12)
Can you please also send me the entire text output of the training script?

**Comment by ngaggion** [comment]
After changing the number of augmentation workers it is still using the full 64GB but thanks to the swap memory the process is not dying, but also not finishing the first epoch. The training script looks like this:

````
Please cite the following paper when using nnUNet:

Isensee, F., Jaeger, P.F., Kohl, S.A.A. et al. "nnU-Net: a self-configuring method for deep learning-based biomedical image segmentation." Nat Methods (2020). https://doi.org/10.1038/s41592-020-01008-z


If you have questions or suggestions, feel free to open an issue at https://github.com/MIC-DKFZ/nnUNet

###############################################
I am running the following nnUNet: 2d
My trainer class is:  <class 'nnunet.training.network_training.nnUNetTrainerV2.nnUNetTrainerV2'>
For that I will be using the following configuration:
num_classes:  2
modalities:  {0: 'S'}
use_mask_for_norm OrderedDict([(0, False)])
keep_only_largest_region None
min_region_size_per_class None
min_size_per_class None
normalization_schemes OrderedDict([(0, 'nonCT')])
stages...

stage:  0
{'batch_size': 1406, 'num_pool_per_axis': [1, 0], 'patch_size': array([3280,    1]), 'median_patient_size_in_voxels': array([2464, 3280,    1]), 'current_spacing': array([1., 1., 1.]), 'original_spacing': array([1., 1., 1.]), 'pool_op_kernel_sizes': [[2, 1]], 'conv_kernel_sizes': [[3, 3], [3, 3]], 'do_dummy_2D_data_aug': False}

I am using stage 0 from these plans
I am using batch dice + CE loss

I am using data from this folder:  nnUNet_preprocessed/Task601_MulticlassRootSegmentation/nnUNetData_plans_v2.1_2D
###############################################
loading dataset
loading all case properties
2022-03-28 08:48:26.559255: Using splits from existing split file: nnUNet_preprocessed/Task601_MulticlassRootSegmentation/splits_final.pkl
2022-03-28 08:48:26.585853: The split file contains 5 splits.
2022-03-28 08:48:26.586107: Desired fold for training: 0
2022-03-28 08:48:26.586288: This split has 103 training and 26 validation cases.
unpacking dataset
done
2022-03-28 08:48:30.086752: lr: 0.01
using pin_memory on device 0
````

**Comment by FabianIsensee** [comment]
Yeah. Oops. Your 2D data is broken! 

'batch_size': 1406
'patch_size': array([3280,    1])
'median_patient_size_in_voxels': array([2464, 3280,    1])

You probably have the wrong axis ordering in the .nii.gz files. `'original_spacing': array([1., 1., 1.])` should also have one large number in it for 2d images.

Please take a look at https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunet/dataset_conversion/Task120_Massachusetts_RoadSegm.py

Best,
Fabian

**Comment by FabianIsensee** [comment]
This might also be useful for interpreting the plans pkl files of 2d data. Your should look similar to this
https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/tutorials/edit_plans_files.md

**Comment by ngaggion** [comment]
Thank you Fabian, I changed my png to .nii.gz script and the problem was solved. 
I'm closing the issue.

Best regards,
Nicolas
